Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm

```diff
@@ -327,25 +456,107 @@ - (void)captureOutput:(AVCaptureOutput *)__unused captureOutput didOutputSampleB
     }
 
     CVPixelBufferRef pixelBuffer{CMSampleBufferGetImageBuffer(sampleBuffer)};
-    id<MTLTexture> textureBGRA{nil};
-
     size_t width = CVPixelBufferGetWidthOfPlane(pixelBuffer, 0);
     size_t height = CVPixelBufferGetHeightOfPlane(pixelBuffer, 0);
-    MTLPixelFormat pixelFormat{MTLPixelFormatBGRA8Unorm};
-    
-    CVMetalTextureRef texture{nullptr};
-    CVReturn status{CVMetalTextureCacheCreateTextureFromImage(nullptr, implData->textureCache, pixelBuffer, nullptr, pixelFormat, width, height, 0, &texture)};
-    if (status == kCVReturnSuccess)
+
+    // Update both metal textures used by the renderer to display the camera image.
+    id<MTLTexture> textureY = [self getCameraTexture:pixelBuffer plane:0];
+    id<MTLTexture> textureCbCr = [self getCameraTexture:pixelBuffer plane:1];
+
+    if(textureY != nil && textureCbCr != nil)
     {
-        textureBGRA = CVMetalTextureGetTexture(texture);
-        CFRelease(texture);
+        @synchronized(self) {
```

Comment:
Seems like all this drawing needs to happen in `UpdateCameraTexture`. The camera is constantly capturing frames, but Babylon's texture is only supposed to be updated when `UpdateCameraTexture` is called (from the call in the JS side to `VideoTexture.update()`). Otherwise, this code is updating Babylon's texture regardless of the call to `VideoTexture.update()`, and when rendering with it it would be updated unexpectedly.

It seems like `UpdateCameraTexture` should recreate the RBGATexture as needed (e.g. if the resolution changes) and do the `overrideInteral` directly, and then do the actual drawing operation. This way, we aren't even doing much work unless/until the BJS side makes a call to `VideoTexture.update()`.

Reply:
This is also really similar to what happens on the Android side.

Reply:
Ya, this seemed like the wrong place to me, too, but when it's in `UpdateCameraTexture` the Metal command buffer errors out after a few minutes with:
```
2022-08-29 14:42:39.689979-0400 Playground[40466:1388338] Execution of the command buffer was aborted due to an error during execution. Caused GPU Timeout Error (00000002:kIOGPUCommandBufferCallbackErrorTimeout)
```
...but it works fine if it's in `didOutputSampleBuffer`.

Reply:
I've got some ideas on what the issue is. Trying a fix now.